In [1]:
import numpy as np
import pandas as pd 
import matplotlib.pyplot as plt 
import seaborn as sns
df = pd.read_csv("data/heart.csv")
df

,age,sex,cp,trestbps,chol,fbs,restecg,thalach,exang,oldpeak,slope,ca,thal,target
0,63,1,3,145,233,1,0,150,0,2.3,0,0,1,1
1,37,1,2,130,250,0,1,187,0,3.5,0,0,2,1
2,41,0,1,130,204,0,0,172,0,1.4,2,0,2,1
3,56,1,1,120,236,0,1,178,0,0.8,2,0,2,1
4,57,0,0,120,354,0,1,163,1,0.6,2,0,2,1
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
298,57,0,0,140,241,0,1,123,1,0.2,1,0,3,0
299,45,1,3,110,264,0,1,132,0,1.2,1,0,3,0
300,68,1,0,144,193,1,1,141,0,3.4,1,2,3,0
301,57,1,0,130,131,0,1,115,1,1.2,1,1,3,0


### سوال 5

از دید کسب‌وکار، هر ستون چه مفهومی دارد؟

(جدول توضیحات تهیه شود.)

In [2]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 303 entries, 0 to 302
Data columns (total 14 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   age       303 non-null    int64  
 1   sex       303 non-null    int64  
 2   cp        303 non-null    int64  
 3   trestbps  303 non-null    int64  
 4   chol      303 non-null    int64  
 5   fbs       303 non-null    int64  
 6   restecg   303 non-null    int64  
 7   thalach   303 non-null    int64  
 8   exang     303 non-null    int64  
 9   oldpeak   303 non-null    float64
 10  slope     303 non-null    int64  
 11  ca        303 non-null    int64  
 12  thal      303 non-null    int64  
 13  target    303 non-null    int64  
dtypes: float64(1), int64(13)
memory usage: 33.3 KB


: بر اساس اسن متد ما 12 ستون داریم که به ترتب


 سن ,جنسیت ,نوع درد قفسه سینه ,فشارخون در حال ,استراحت  میزان کلسترول خون ,قند خون ناشتا ,نتیجه نوار قلب 
 
 ,حداکثر ضربان قلب ,درد ناشی از ورزش ,افت سیتی در تست ورزش ,شیب سیتی در تست ورزش ,تعداد رگ های اصلی ,نتیجه تست تالاسمی ,وضعیت بیماری قلبی

In [4]:
df.head()

,age,sex,cp,trestbps,chol,fbs,restecg,thalach,exang,oldpeak,slope,ca,thal,target
0,63,1,3,145,233,1,0,150,0,2.3,0,0,1,1
1,37,1,2,130,250,0,1,187,0,3.5,0,0,2,1
2,41,0,1,130,204,0,0,172,0,1.4,2,0,2,1
3,56,1,1,120,236,0,1,178,0,0.8,2,0,2,1
4,57,0,0,120,354,0,1,163,1,0.6,2,0,2,1


In [3]:
df.describe()

,age,sex,cp,trestbps,chol,fbs,restecg,thalach,exang,oldpeak,slope,ca,thal,target
count,303.000000,303.000000,303.000000,303.000000,303.000000,303.000000,303.000000,303.000000,303.000000,303.000000,303.000000,303.000000,303.000000,303.000000
mean,54.366337,0.683168,0.966997,131.623762,246.264026,0.148515,0.528053,149.646865,0.326733,1.039604,1.399340,0.729373,2.313531,0.544554
std,9.082101,0.466011,1.032052,17.538143,51.830751,0.356198,0.525860,22.905161,0.469794,1.161075,0.616226,1.022606,0.612277,0.498835
min,29.000000,0.000000,0.000000,94.000000,126.000000,0.000000,0.000000,71.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000
25%,47.500000,0.000000,0.000000,120.000000,211.000000,0.000000,0.000000,133.500000,0.000000,0.000000,1.000000,0.000000,2.000000,0.000000
50%,55.000000,1.000000,1.000000,130.000000,240.000000,0.000000,1.000000,153.000000,0.000000,0.800000,1.000000,0.000000,2.000000,1.000000
75%,61.000000,1.000000,2.000000,140.000000,274.500000,0.000000,1.000000,166.000000,1.000000,1.600000,2.000000,1.000000,3.000000,1.000000
max,77.000000,1.000000,3.000000,200.000000,564.000000,1.000000,2.000000,202.000000,1.000000,6.200000,2.000000,4.000000,3.000000,1.000000


## : بر اساس این متد متوجه شدم  

همه سطر ها داده داد و رنج سن بیماران بین 39 سال تا 77 سال هستش میانگینشون 54 سال است و انحراف معیارش 9 پس داده ها پراکندگی کمی دارند 

فشار خون در حال استراحت بیماران بین 94 تا 20 هستس که با توجه به میانگین بیشتر بیماران و چارک ها فشارخون نرمالی دارند 

کلسترول بیماران بین 126 تا 564 هستش که انحراف معیار 54 بهمون میگه پراکندگی زیادی داریم و این تفاوت بالای ماکسیمم و مینیمم نشان دهنده پر خطر بودن مقادیر بالا هستش

ستون حداکثر ضربان قلب بازه 71 تا 202 نشون میده که میانگین 149 ضربه در دقیقه است واین ستون اطلاعات ارزشمند پزشکی نشون میده چون چارک ها با ماکسیمم و مینیمم خیلی فاصله دارندکه میتونه نشونه مشکل جدی بعضی از بیماران باشه

 ستون میزان افت سی تی میانگین 1.4 داره که یعنی بیشتر بیماران افت کمی دارند و چارک اول نشون دهنذه اینه یک چهارم بیماران هیچ افتی نداشتند

In [6]:
(df["sex"].value_counts()/ len(df))*100

sex
1    68.316832
0    31.683168
Name: count, dtype: float64

درصد بیماران مرد و68 


31 درصد بیماران زن هستند 

In [7]:
df["cp"].value_counts() 

cp
0    143
2     87
1     50
3     23
Name: count, dtype: int64

باتوجه به این جدول 143نفر درد معمولی دارند که به قلب مرتبط نیست و87 نفر درد غیر معمول دارند و 50 نفر دردی دارند که به قلب مرتبط و 23 نفر درد شدید قفسه سینه دارند 

In [10]:
pd.crosstab(df["sex"], df["cp"],normalize="index")*100

cp,0,1,2,3
sex,,,,
0,40.625000,18.750000,36.458333,4.166667
1,50.241546,15.458937,25.120773,9.178744


با توجه به این جدول 4 درصد خانم ها و 9 درصد اقایونی که به دکتر مراجعه کردند درد شدید داشتند 

In [14]:
pd.crosstab( df["fbs"], df["target"],normalize="index")*100

target,0,1
fbs,,
0,44.961240,55.038760
1,48.888889,51.111111


باتوجه به این جدول 55 درصد کسایی که قند خون ندارند مشکل قلبی دارند و 51 درصد کسایی که قند خون دارند مشکل قلبی هم دارند تقریبا میشه گفت قند خون باعث بیماری قلبی نمیشود

In [15]:
df["restecg"].value_counts()

restecg
1    152
0    147
2      4
Name: count, dtype: int64

نوار قلب در حال استراحت از مراجع کنندگان تهیه شده که نتیجه 147 نفر طبیعی بودهو 152 نفر ناهنجاری خفیف موج اس تی تی داشتند و 4 نفر احتما بزرگ شدن بطن چپ داشتند

In [18]:
df["exang"].value_counts(normalize=True)*100

exang
0    67.326733
1    32.673267
Name: proportion, dtype: float64

درصد 67 هنگام ورزش دردی در قفسه سینه نداشتند ولی 32 درصد درد داشتند

In [19]:
df["slope"].value_counts()


slope
2    142
1    140
0     21
Name: count, dtype: int64

پاسخ قلب به ورزش 21 نفر شیب نزولی داشته و برای 140 نفر شیب صاف و 142 نفر شیب صعودی

In [22]:
df["ca"].value_counts()

ca
0    175
1     65
2     38
3     20
4      5
Name: count, dtype: int64

از 303 تست دهنده 175 نفر رگ مشخص در انژیو نداشتند و65 نفر یک رگ و38 نفر 2 رگ . 20 نفر 3 رگ و5 نفر 4 رگ

In [25]:
df["thal"].value_counts()

thal
2    166
3    117
1     18
0      2
Name: count, dtype: int64

دو نفر از مراجعه کنندگان کاملا سالم بوذن و 18 نفر نقص قابل برگشت دارند و 166 نفر یک بیماری ثابت دارند و 177 نفر بیمار غیر طبیعی بوذن و احتمال خیلی زیادی مشکل جدی دارند

In [26]:
df["target"].value_counts()

target
1    165
0    138
Name: count, dtype: int64

 بیشتر افراد  بیماری قلبی دارند یعنی 165 نفر ولی 138 نفر بیمار نیستند